### ISSUE TITLE
temporary schedules: can't view shifts when rotation with 0-1 participants is assigned

### ISSUE THREAD

**Issue by Forfold** [target/goalert#960]
**Describe the Bug:**
If a rotation with 0 or 1 participant is assigned to a schedule, the page breaks. This does not seem to occur if more than 1 participant is on a rotation.

**Steps to Reproduce:**
1. Create an empty rotation (or a rotation with 1 participant)
2. Create a schedule
3. Add the rotation to the schedule
4. Attempt to visit the shifts page via the quick links section

**Additional Context:**
This appears to be a backend bug. When querying from the graphql playground, `users` is returned as `null` and `userID` is set to the rotation's ID. This causes the UI to break, as the API does not expect the user object to ever be null if shifts are provided.

**Screenshots/Stack Traces:**
<img width="550" src="https://user-images.githubusercontent.com/11381794/96903759-17d6fc00-144b-11eb-89e6-2dc73758141e.png" />
_Note: The input start/end date used above was the following:_
```
shifts(start:"2020-11-20T00:00:00.000Z" end:"2020-11-30T00:00:00.000Z")
```

**Application Version:**
`feature/temp-schedules` branch

**Comment by mastercactapus** [comment]
oooh, that's interesting. the most likely cause for something like that should be somewhere we're calling `Target.TargetID()` without checking it's type

Maybe add a conditional inside `singlerulecalcullator.go`

![image](https://user-images.githubusercontent.com/595010/96904144-57f2aa80-145c-11eb-8aa2-3fd6f5fef6bc.png)
 
The code assumes that if it doesn't have a rotation prop it must be a user, but perhaps empty rotations are never initialized (for performance).

**Comment by mastercactapus** [comment]
^ pretty sure that's our culprit -- reviewing the code the on-call package get's the list of rotations with a `JOIN` against `rotation_state` -- if there are zero participants rotation state isn't populated, so rotations with no participants will always equate to `nil` down the line

**Comment by mastercactapus** [comment]
Fixed by #962
